### PR TITLE
CIO-459 copy highlighted branch name to clipboard

### LINKED ISSUE #459: Hotkey to copy highlighted branch name to clipboard
As a user, I would like to be able to navigate through CIO, and press a hot key to copy the current highlighted branch name to my clipboard.

### PR DESCRIPTION
**Pull request by mahendraHegde**
Fixes #459 

## Description
copy highlighted branch name to clipboard.
used key `y` for copying(vim-style)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My commits follow the [commitizen](https://github.com/commitizen/cz-cli#philosophy) commit convention
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -297,6 +298,18 @@ export const start = async args => {
     screen.spawn('git', ['log', args.join('/'), ...gitLogArguments]);
   });
 
+  branchTable.key('y', function() {
+    const selection = parseSelection(this.items[this.selected].content);
+    const gitBranch = selection[2];
+    copyToClipBoard(gitBranch)
+      .then(() => {
+        logger.log(`branch name ${gitBranch} copied.`);
```
Comment (kienD) [inline src/app.js]: Can you capitalize the first character of this sentence?

File: src/app.js [hunk 2]
```diff
@@ -297,6 +298,18 @@ export const start = async args => {
...
+      })
+      .catch(error => {
+        logger.error(`unable to copy. ${JSON.stringify(error)}`);
```
Comment (kienD) [inline src/app.js]: ^

File: README.md [hunk 3]
```diff
@@ -129,6 +129,7 @@ See a git log for the highlighted ref by pressing <kbd>SPACE</kbd>
 | <kbd>n</kbd>                                                     | Jump to next search result                          |
 | <kbd>N</kbd>                                                     | Jump to previous search result                      |
 | <kbd>q</kbd>, <kbd>Ctrl</kbd>+<kbd>c</kbd>, <kbd>esc</kbd>       | Quit                                                |
+| <kbd>y</kbd>      | Copy highlighted branch name                                                		|
```
Comment (kienD) [inline README.md]: Can you format this to be like the other lines?

File: src/app.js [hunk 4]
```diff
@@ -297,6 +298,18 @@ export const start = async args => {
     screen.spawn('git', ['log', args.join('/'), ...gitLogArguments]);
   });
 
+  branchTable.key('y', function() {
+    const selection = parseSelection(this.items[this.selected].content);
+    const gitBranch = selection[2];
+    copyToClipBoard(gitBranch)
+      .then(() => {
```
Comment (kienD) [inline src/app.js]: Can we use `async` and `await` to match the pattern of the other functions here instead?

File: README.md (hunk 3 above)
Comment (jwu910) [reply README.md]: Can we also move this up between the command `enter` and `space` and follow the verbiage of "Copy highlighted item"?

File: src/app.js (hunk 1 above)
Comment (mahendraHegde) [reply src/app.js]: done.

File: src/app.js (hunk 2 above)
Comment (mahendraHegde) [reply src/app.js]: done.

File: README.md (hunk 3 above)
Comment (mahendraHegde) [reply README.md]: done.

File: src/app.js (hunk 4 above)
Comment (mahendraHegde) [reply src/app.js]: done.

**Comment by jwu910** [comment]
Just started reviewing :)

File: src/app.js [hunk 5]
```diff
@@ -297,6 +298,17 @@ export const start = async args => {
...
   });
 
+  branchTable.key('y', async function() {
+    const selection = parseSelection(this.items[this.selected].content);
+    const gitBranch = selection[2];
+    try {
+      await copyToClipBoard(gitBranch);
+      logger.log(`Branch name ${gitBranch} copied.`);
+    } catch (error) {
+      logger.error(`Unable to copy. ${JSON.stringify(error)}`);
```
Comment (jwu910) [inline src/app.js]: Can we swap the period for a colon here? I think that should be fine.

Comment (mahendraHegde) [reply src/app.js]: thanks for the review @jwu910  @kienD , I've modified suggested changes, plz review.

File: src/app.js [hunk 6]
```diff
@@ -297,6 +298,17 @@ export const start = async args => {
     screen.spawn('git', ['log', args.join('/'), ...gitLogArguments]);
   });
 
+  branchTable.key('y', async function() {
+    const selection = parseSelection(this.items[this.selected].content);
+    const gitBranch = selection[2];
+    try {
+      await copyToClipBoard(gitBranch);
+      logger.log(`Branch name ${gitBranch} copied.`);
```
[the remark this answers is not in the document]
Comment (jwu910) [reply src/app.js]: Thanks for changes! Think we missed a small detail on this comment and I wanted to address it before we merged.

Can you wrap the variable injection in quotes so the log will show a message like this:
```[log] "my-branch-name" copied to clipboard```

This should make it a little easier to see.

Thanks!

**Comment by jwu910** [comment]
@mahendraHegde Just one minor tweak and we should be good to go!

File: src/app.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (mahendraHegde) [reply src/app.js]: done

**Comment by kienD** [comment]
Thanks for your contribution, @mahendraHegde!

**Pull request merged**